src/Analyzers/CSharp/Analyzers/UseUTF8StringLiteral/UseUTF8StringLiteralDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,150 @@
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Buffers;
+using System.Collections.Immutable;
+using System.Data;
+using System.Linq;
+using System.Text;
+using Microsoft.CodeAnalysis.CodeStyle;
+using Microsoft.CodeAnalysis.CSharp.CodeStyle;
+using Microsoft.CodeAnalysis.CSharp.Extensions;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Operations;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.UseUTF8StringLiteral
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    internal sealed partial class UseUTF8StringLiteralDiagnosticAnalyzer : AbstractBuiltInCodeStyleDiagnosticAnalyzer
+    {
+        public const string StringValuePropertyName = "StringValue";
+
+        public UseUTF8StringLiteralDiagnosticAnalyzer()
+            : base(IDEDiagnosticIds.UseUTF8StringLiteralDiagnosticId,
+                EnforceOnBuildValues.UseUTF8StringLiteral,
+                CSharpCodeStyleOptions.PreferUTF8StringLiteral,
+                LanguageNames.CSharp,
+                new LocalizableResourceString(nameof(CSharpAnalyzersResources.Convert_to_UTF8_string_literal), CSharpAnalyzersResources.ResourceManager, typeof(CSharpAnalyzersResources)),
+                new LocalizableResourceString(nameof(CSharpAnalyzersResources.Use_UTF8_string_literal), CSharpAnalyzersResources.ResourceManager, typeof(CSharpAnalyzersResources)))
+        {
+        }
+
+        protected override void InitializeWorker(AnalysisContext context)
+            => context.RegisterCompilationStartAction(context =>
+            {
+                if (context.Compilation.LanguageVersion() < LanguageVersion.Preview)
+                    return;
+
+                var expressionType = context.Compilation.GetTypeByMetadataName(typeof(System.Linq.Expressions.Expression<>).FullName!);
+
+                context.RegisterOperationAction(c => AnalyzeOperation(c, expressionType), OperationKind.ArrayCreation);
+            });
+
+        private void AnalyzeOperation(OperationAnalysisContext context, INamedTypeSymbol? expressionType)
+        {
+            var arrayCreationExpression = (IArrayCreationOperation)context.Operation;
+
+            // Don't offer if the user doesn't want it
+            var option = context.GetOption(CSharpCodeStyleOptions.PreferUTF8StringLiteral);
+            if (!option.Value)
+                return;
+
+            // Only replace arrays with initializers
+            if (arrayCreationExpression.Initializer is null)
+                return;
+
+            var byteType = context.Compilation.GetSpecialType(SpecialType.System_Byte);
+            var elementType = (arrayCreationExpression.Type as IArrayTypeSymbol)?.ElementType;
+            if (!SymbolEqualityComparer.Default.Equals(elementType, byteType))
```

Comment:
I may get the syntax wrong, but the idea was to avoid unnecessary `GetSpecialType` and symbol comparison and just `SpecialType` property directly.

```suggestion
            if (arrayCreationExpression.Type is IArrayTypeSymbol { ElementType.SpecialType: SpecialType.System_Byte })
```